src/roi/Roi.ts

```diff
@@ -390,25 +405,37 @@ function getPerimeterInfo(roi: Roi) {
         let nbAround = 0;
         if (column === 0) {
           nbAround++;
-        } else if (roi.externalIDs.includes(data[target - 1])) {
+        } else if (
+          roi.externalBorders.find((el) => el.connectedID === data[target - 1])
+        ) {
           nbAround++;
         }
 
         if (column === roiMap.width - 1) {
           nbAround++;
-        } else if (roi.externalIDs.includes(data[target + 1])) {
+        } else if (
+          roi.externalBorders.find((el) => el.connectedID === data[target + 1])
```

Comment:
Can you revert to not using `includes`?

You can recreate the `externalIDs` array before the loop.

Reply:
Not sure I understand. What do you mean by "revert to not using 'includes'"? It is not used anymore. Or you want me to revert it to use it again?